File: CalculatorJava/src/main/java/com/softuni/controller/HomeController.java
package com.softuni.controller;

import com.softuni.models.Calculator;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class HomeController {
    @GetMapping("/")
    public String index(Model model) {
        model.addAttribute("operator", "+");
        model.addAttribute("view", "views/calculatorForm");
        return "base-layout";
    }

    @PostMapping("/")
    public String calculate(@RequestParam String leftOperand, @RequestParam String rightOperand, @RequestParam String operator, Model model) {
        double fNum;
        double sNum;

        try {
            fNum = Integer.parseInt(leftOperand);
        } catch (NumberFormatException e) {
            fNum = 0;
        }
        try {
            sNum = Integer.parseInt(rightOperand);
        } catch (NumberFormatException e) {
            sNum = 0;
        }

        Calculator calc = new Calculator(fNum, sNum, operator);
        double res = calc.calculate();

        model.addAttribute("leftOperand", calc.getLeftOperand());
        model.addAttribute("operator", calc.getOperator());
        model.addAttribute("rightOperand", calc.getRightOperand());

        model.addAttribute("result", res);

        model.addAttribute("view", "views/CalculatorForm");
        return "base-layout";
    }
}

File: CalculatorCSharp/Calculator-CSharp/Controllers/HomeController.cs
using Calculator_CSharp.Models;
using System;
using System.Web.Mvc;

namespace Calculator_CSharp.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index(Calculator calculator)
        {
            return View(calculator);
        }

        [HttpPost]
        public ActionResult Calculate(Calculator calculator)
        {
            calculator.result = CalculateResult(calculator);
            return RedirectToAction("Index", calculator);
        }
        public double CalculateResult(Calculator calculator)
        {
            double res = 0;

            switch (calculator.operand)
            {
                case "+":
                    res = calculator.leftOperand + calculator.rightOperand;
                    break;
                case "-":
                    res = calculator.leftOperand - calculator.rightOperand;
                    break;
                case "*":
                    res = calculator.leftOperand * calculator.rightOperand;
                    break;
                case "/":
                    res = calculator.leftOperand / calculator.rightOperand;
                    break;
                case "^":
                    res = Math.Pow(calculator.leftOperand, calculator.rightOperand);
                    break;
                case "||":
                    if ((calculator.leftOperand == 0 || calculator.leftOperand == 1) && (calculator.rightOperand == 0 || calculator.rightOperand == 1))
                    {
                        if (calculator.leftOperand == 0 && calculator.rightOperand == 0)
                        {
                            res = 0;
                        }
                        else
                            res = 1;
                    }
                    else
                        res = -1;//invalid input
                    break;
                case "&&":
                    if ((calculator.leftOperand == 0 || calculator.leftOperand == 1) && (calculator.rightOperand == 0 || calculator.rightOperand == 1))
                    {
                        if (calculator.leftOperand == 1 && calculator.rightOperand == 1)
                        {
                            res = 1;
                        }
                        else
                            res = 0;
                    }
                    else
                        res = -1;//invalid input
                    break;
            }

            return res;
        }
    }
}